pilot/pkg/monitoring/monitoring.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitoring
+
+import (
+	"context"
+	"fmt"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type (
+	// Metric collects numerical observations. These observations are aggregated and exported
+	// to OpenCensus views.
+	Metric interface {
+
+		// Increment records a value of 1 for the current measure. For view.Count and view.Sum
+		// aggregations, this is equivalent to adding 1 to the current value.
+		Increment()
+
+		// Name returns the name value of a Metric
+		Name() string
+
+		// Record makes an observation of the provided value for the given measure.
+		Record(value float64)
+
+		// WithTags creates a new Metric, with the tag mutations provided. This allows creating
+		// a set of pre-dimensioned data for recording and export purposes.
+		WithTags(tags ...tag.Mutator) Metric
+
+		// View returns the OpenCensus view structure to which this Metric is being exported.
+		View() *view.View
+	}
+
+	// MetricOpts provides the basic options for a Metric.
+	MetricOpts struct {
```

Comment:
@mandarjog is this worthwhile? I used this to pass some other Opts at first, but now it is just {Name, Description}. We could get rid of it and change the `New*` methods to take `name, description string` args instead.

Reply:
i've gone ahead and removed.